libsel4/include/sel4/bootinfo_types.h

```diff
@@ -86,11 +86,14 @@ typedef struct seL4_BootInfoHeader {
 
 /* Bootinfo identifiers share a global namespace, even if they are arch or platform specific
  * and are enumerated here */
-#define SEL4_BOOTINFO_HEADER_PADDING 0
-#define SEL4_BOOTINFO_HEADER_X86_VBE 1
-#define SEL4_BOOTINFO_HEADER_X86_MBMMAP 2
-#define SEL4_BOOTINFO_HEADER_X86_ACPI_RSDP 3
-#define SEL4_BOOTINFO_HEADER_X86_FRAMEBUFFER 4
-#define SEL4_BOOTINFO_HEADER_X86_TSC_FREQ 5 // frequency is in mhz
-#define SEL4_BOOTINFO_HEADER_FDT 6
-#define SEL4_BOOTINFO_HEADER_NUM SEL4_BOOTINFO_HEADER_FDT + 1
+typedef enum {
+    SEL4_BOOTINFO_HEADER_PADDING            = 0,
+    SEL4_BOOTINFO_HEADER_X86_VBE            = 1,
+    SEL4_BOOTINFO_HEADER_X86_MBMMAP         = 2,
+    SEL4_BOOTINFO_HEADER_X86_ACPI_RSDP      = 3,
+    SEL4_BOOTINFO_HEADER_X86_FRAMEBUFFER    = 4,
+    SEL4_BOOTINFO_HEADER_X86_TSC_FREQ       = 5, /* frequency is in MHz */
+    SEL4_BOOTINFO_HEADER_FDT                = 6, /* device tree */
```

Comment:
If only the SEL4_BOOTINFO_HEADER* are used as constants, probably  typedef and seL4_BootInfoID are not useful. BTW, what does seL4_BootInfoID mean?

Reply:
This is used for the field `ìd` of `seL4_BootInfoHeader`. We could drop the name`seL4_BootInfoID` and make the enum anonymous also, might even be better as the type is unused

Reply:
I'd rather avoid anonymous types. If this ever comes up in verification, we'd need some name to refer to it and relate it to other things. It currently doesn't occur in verification and so wouldn't be an immediate problem, but as a general tendency it's preferable for types to have names.